### ISSUE TITLE
Job build takes a long time to start when Vault is not available

### ISSUE THREAD

**Issue by ralekseenkov** [concourse/concourse#3331]
## Bug Report

If previously "+" button would instantly spin up a new build, now we are seeing CreateJobBuild takes exactly 5 seconds. You can see that in Chrome Developer Console, as well as in Grafana.

## Steps to Reproduce

Basically happens on any pipeline.

## Version Info

* Concourse version: 5.0 RC
* Deployment type (BOSH/Docker/binary): Kubernetes
* Infrastructure/IaaS: Azure AKS
* Browser (if applicable): Chrome
* Did this used to work? Yes

If there is a way to profile it and capture some tracing info to see what takes long time, we are happy to help figure this out.

**Comment by ralekseenkov** [comment]
<img width="1202" alt="screen shot 2019-02-19 at 9 38 28 am" src="https://user-images.githubusercontent.com/609325/53035702-e2cf0680-342a-11e9-94e7-d5addd7a86ea.png">

screenshot from Grafana, showing 5 sec API calls for every new job build

**Comment by vito** [comment]
Shot in the dark: could be faulty DNS configuration? iirc 5 seconds is a common default for DNS timeouts, after which it'll attempt the next server. What's in your `/etc/resolv.conf` on the workers? Could you try explicitly configuring DNS servers (e.g. `8.8.8.8`)?

This would explain why both `CreateJobBuild` and general resource checking feels slower, as they both perform resource checks.

fwiw, I don't see this slowness on our instance, so it appears to be deployment-specific.

**Comment by metacoma** [comment]
> Shot in the dark: could be faulty DNS configuration? iirc 5 seconds is a common default for DNS timeouts, after which it'll attempt the next server. What's in your `/etc/resolv.conf` on the workers? Could you try explicitly configuring DNS servers (e.g. `8.8.8.8`)?

Looks like problem not in DNS configuration, same 5s timeout with 8.8.8.8 server.

![5s](https://user-images.githubusercontent.com/5146707/53039300-55c29680-3498-11e9-8d99-d1703222d969.png)

**Comment by vito** [comment]
@metacoma Hm, I believe you but just to further confirm could you try intercepting into one of the containers and seeing how long it takes to e.g. `curl https://google.com` (or some other location)?

Is there anything interesting in the logs (with `--log-level debug`) when this happens?

**Comment by ralekseenkov** [comment]
@vito 

I think we found something.

Disabled vault integration... and Concourse is blazing fast now.

The summary is:
* Our instance of Vault was not working correctly, always responding with 400 to all lookups
* All of the pipelines have parameters explicitly specified and were actually not supposed to go to vault
* What likely caused the slowness is automatic retries of variables that was introduced in 5.0 (SecretRetryConfig default is 5 attempts and 1 second delay)

The big question for me is why is it trying to reach out to Vault. I.e. what exactly happens when CreateJobBuild runs and a new build is scheduled... 😮😮😮

P.S. There is another related issue https://github.com/concourse/concourse/pull/3334. I believe we need to get this patch merged very soon. It likely won't solve the issue completely, but it will be easier for me to continue looking, once that PR is merged

**Comment by ralekseenkov** [comment]
@vito can the reason be that, even though all vars are interpolated, Concourse is somehow trying to call List() on RetryableVariables, adding the delay?

**Comment by cirocosta** [comment]
Nice findings, @ralekseenkov! 

Regarding pin-pointing that it was related to credential management, it feels to me that maybe having some metrics around the resolution of such values could help 🤔 One would probably be able to notice that in a quicker way at least.

I'll write a separate issue for it as I've seen some failures in this area myself and such metrics would help me too

**Comment by ralekseenkov** [comment]
@vito turns out this is the problem:
https://github.com/concourse/concourse/blob/master/atc/creds/vault/vault_factory.go#L31

Concourse silently swallows timeouts when connecting to Vault, making Concourse UI/API quite slow.

It slows down everything that calls NewVariables - creating builds, listing job inputs, resource scanners, etc. So the whole Concourse becomes very sluggish when Vault goes down.

I will look into options of getting this fixed
